code/__HELPERS/trait_helpers.dm

```diff
@@ -127,7 +127,7 @@
 		: FALSE)
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits ? (target.status_traits[trait] ? (length(target.status_traits[trait] - source) > 0) : FALSE) : FALSE)
 /// A simple helper for checking traits in a mob's mind
-#define HAS_MIND_TRAIT(target, trait) (HAS_TRAIT(target, trait) || (target.mind ? HAS_TRAIT(target.mind, trait) : FALSE))
+#define HAS_MIND_TRAIT(target, trait) ((target && target.mind && HAS_TRAIT(target, trait)) || (target && target.mind && target.mind.status_traits ? HAS_TRAIT(target.mind, trait) : FALSE))
```

Comment:
```suggestion
#define HAS_MIND_TRAIT(target, trait) ((target?.mind ? HAS_TRAIT(target.mind, trait) : FALSE))
```
I think we don't need the first HAS_TRAIT at all. The define should be for explicitly checking if the mind has a trait, in my opinion.

Reply:
We did need it, so you could call HAS_MIND_TRAIT on minds themselves